src/Umbraco.Core/Extensions/StringExtensions.cs

```diff
@@ -57,17 +58,24 @@ static StringExtensions()
     /// <returns></returns>
     public static int[] GetIdsFromPathReversed(this string path)
     {
-        string[] pathSegments = path.Split(Constants.CharArrays.Comma, StringSplitOptions.RemoveEmptyEntries);
-        List<int> nodeIds = new(pathSegments.Length);
-        for (int i = pathSegments.Length - 1; i >= 0; i--)
+        ReadOnlySpan<char> pathSpan = path.AsSpan();
+        List<int> nodeIds = [];
+        foreach (Range rangeOfPathSegment in pathSpan.Split(Constants.CharArrays.Comma))
```

Comment:
I will note the removal of `StringSplitOptions.RemoveEmptyEntries` but as we're doing `int.TryParse` on these entries anyway, those would return `false` so all good there anyway.